mobile_verifier/src/reward_shares.rs

```diff
@@ -203,16 +203,38 @@ pub fn dc_to_mobile_bones(dc_amount: Decimal, mobile_bone_price: Decimal) -> Dec
         .round_dp_with_strategy(DEFAULT_PREC, RoundingStrategy::ToPositiveInfinity)
 }
 
+#[derive(Default, Debug)]
+pub struct Share {
+    cell_type_weight: Decimal,
+    speed_multiplier: Decimal,
+    aggregated_location_trust_weight: Decimal,
+    aggregated_hb_count: Decimal,
+}
+
+impl Share {
+    fn update(&mut self, heartbeat_reward: HeartbeatReward, speed_multiplier: Decimal) {
+        self.cell_type_weight = heartbeat_reward.cell_type_weight;
```

Comment:
seems like the `cell_type_weight` shouldn't change once set. If this single `update` method is supposed to act as an upsert operation that field should probably be an `cell_type_weight: Option<Decimal>` and then the `update` method checks and only sets the value once if it's not already set.

The perpetual assignment implies this value should actually change with the input of each new heartbeat when I don't think that makes sense; the weight should be stable for the life of the radio.

Reply:
Call type weight and speed multiplier will not change.  They are located here for simplicity...the values are stable.  They will not change across updates.  I dont think wrapping each in an Option provides any real additional value and just makes the code more verbose.  Ideally they could be moved out of the Share struct and into the overall RadioShares struct but that complicates things further.  
The best thing to do here is see how this all gets refactored as part of making it compatible with modelled coverage v3 and see if a better home can be found then